src/nwbinspector/_formatting.py

```diff
@@ -75,14 +76,14 @@ class MessageFormatter:
 
     def __init__(
         self,
-        messages: list[InspectorMessage],
+        messages: list[Optional[InspectorMessage]],
         levels: list[str],
         reverse: Optional[list[bool]] = None,
         detailed: bool = False,
         formatter_options: Optional[FormatterOptions] = None,
-    ):
+    ) -> None:
         self.nmessages = len(messages)
-        self.nfiles = len(set(message.file_path for message in messages))
+        self.nfiles = len(set(message.file_path for message in messages))  # type: ignore
```

Comment:
These types are skips however are a weird false positive

We have to annotate various upstream signatures as `list[Optional[InspectorMessage]]` because a given check might have returned `None` - but the way we iterate the generator, those just don't show up in the final list of returned messages so these 'post-inspection' helpers really are typed for `list[InspectorMessage]`